### ISSUE TITLE
Could NOT attach to process

### ISSUE THREAD

**Issue by hamad12a** [WebFreak001/code-debug#371]
I want to attach a shared library (.so) file to a running process.
my `launch.json` file reads,
```
{
        "type": "gdb",
        "request": "attach",
        "name": "Attach to PID",
        "target": "14823",
        "cwd": "${workspaceRoot}",
        "valuesFormatting": "parseText"
    },
```
The output from the debug console,
```
[Inferior 1 (process 18766) detached]
Warning: 'set target-async', an alias for the command 'set mi-async', is deprecated.
Use 'set mi-async'.
[New LWP 18769]
[New LWP 18770]
[New LWP 18771]
[New LWP 18772]
[New LWP 18774]
[New LWP 18775]
[New LWP 18776]
[New LWP 18777]
[New LWP 18781]
[New LWP 18782]
[New LWP 18784]
[New LWP 18786]
[New LWP 18792]
[New LWP 18793]
[New LWP 18794]
[New LWP 18795]
[New LWP 18796]
[New LWP 18797]
[New LWP 18798]
[New LWP 18799]
[New LWP 18800]
[New LWP 18801]
[New LWP 18803]
[New LWP 18804]
[New LWP 18805]
[New LWP 18810]
[New LWP 18811]
[New LWP 18812]
[New LWP 18813]
[New LWP 18816]
[New LWP 18819]
[New LWP 18820]
[New LWP 18821]
[New LWP 18822]
warning: Could not load shared library symbols for bin/linux64/launcher_client.so.
Do you need "set solib-search-path" or "set sysroot"?

[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
0x00007f7a856bd868 in __GI___clock_nanosleep (clock_id=clock_id@entry=0, flags=flags@entry=0, req=0x7ffea540f510, rem=0x7ffea540f500) at ../sysdeps/unix/sysv/linux/clock_nanosleep.c:78
78	../sysdeps/unix/sysv/linux/clock_nanosleep.c: No such file or directory.

```
Additionally, an error shows in the bottom right of VS Code, `Failed to get Stack Trace: Selected thread is running. (from stack-info-depth --thread 1)` , and the process does not freeze after running `launch.json`.

**Comment by GitMensch** [comment]
... so:

* Does it work with plain GDB (`gdb -p 14823`)?
* Do you have breakpoints set where you want to stop?

**Comment by hamad12a** [comment]
@GitMensch 
```
gdb -p 14823
Attaching to process 20897
[New LWP 20898]
[New LWP 20899]
[New LWP 20900]
[New LWP 20901]
[New LWP 20903]
[New LWP 20904]
[New LWP 20905]
[New LWP 20906]
[New LWP 20911]
[New LWP 20912]
[New LWP 20913]
[New LWP 20915]
[New LWP 20925]
[New LWP 20926]
[New LWP 20927]
[New LWP 20929]
[New LWP 20930]
[New LWP 20931]
[New LWP 20932]
[New LWP 20933]
[New LWP 20934]
[New LWP 20935]
[New LWP 20937]
[New LWP 20940]
[New LWP 20941]
[New LWP 20942]
[New LWP 20943]
[New LWP 20944]
[New LWP 20945]
[New LWP 20948]
[New LWP 20951]
[New LWP 20952]
[New LWP 20953]
[New LWP 20954]

warning: Could not load shared library symbols for bin/linux64/launcher_client.so.
Do you need "set solib-search-path" or "set sysroot"?
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
__memmove_avx_unaligned_erms () at ../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S:708
708     ../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S: No such file or directory.
```
Looks attached with command.

**Comment by GitMensch** [comment]
yes, it is attached in both cases. GDB by default stops after attaching, NativeDebug doesn't; if you want that (instead of stopping at a breakpoint you set before) add `"stopAtConnect": true` to you launch.json. While this is documented in the README it seems to only belong to gdbserver, which I think is not the case.

**Comment by hamad12a** [comment]
@GitMensch Perfect, it worked!
Another issue, after setting a breakpoint from VS Code editor source file, at main:10
it says
```
No source file named /home/mine/Downloads/Osiris-master-2/Source/main.cpp.
```
that's might be because I still did not inject `.so` file, okay, I inject using `call (void*)dlopen("/home/mine/Downloads/Osiris-master-2/build/libOsiris.so",  2)` which then returns `$1 = (void *) 0x6257fe00`. 
What to do next to hit the breakpoint?

**Comment by GitMensch** [comment]
That's a bit strange. I'd try adding `"autorun": [ "set breakpoint pending on",]` to launch configuration.

**Comment by hamad12a** [comment]
@GitMensch `"autorun": [ "set breakpoint pending on" ]` answers with yes permanently. i.e., no more questions about pending breakpoints.

**Comment by GitMensch** [comment]
you can execute that statement in the console before / after setting breakpoints - if that is what you want, too

**Comment by hamad12a** [comment]
@GitMensch however, an error is thrown after hitting the first breakpoint
```
The program being debugged stopped while in a function called from GDB. Evaluation of the expression containing the function (dlopen) will be abandoned. When the function is done executing, GDB will silently stop. (from interpreter-exec --thread 1 --frame 0 console "call (void*)dlopen(\"/home/mine/Downloads/Osiris-master-2/build/libOsiris.so\", 2)")
```

**Comment by GitMensch** [comment]
... and that's what GDB says on the command line, too - isn't it?

**Comment by hamad12a** [comment]
I didn't try. I will do

**Comment by GitMensch** [comment]
I don't know; the favorite search machine of yours may help finding out how to do this in GDB

**Comment by hamad12a** [comment]
I close this issue as resolved.

**Comment by hamad12a** [comment]
I close this issue as resolved.

**Comment by GitMensch** [comment]
Out of interest: Did you found a way to do what you want? (that's a GDB issue, but still)

**Comment by hamad12a** [comment]
Indeed. It worked as discussed above; the first breakpoint I mentioned throws that error weirdly by either method, (a bug in the library? who knows). Other breakpoints do not.
